File: scripts/search_secrets.py
import os

def search_files(directory):
    for root, dirs, files in os.walk(directory):
        if 'node_modules' in dirs:
            dirs.remove('node_modules')
        if '.git' in dirs:
            dirs.remove('.git')
        for file in files:
            path = os.path.join(root, file)
            try:
                with open(path, 'r', encoding='utf-8', errors='ignore') as f:
                    content = f.read()
                    if 'service_role' in content or 'SUPABASE_SERVICE_ROLE' in content or 'postgresql://' in content:
                        print(f"Found secret in: {path}")
            except Exception as e:
                pass

search_files('.')

File: scripts/search_secrets.js
import fs from 'fs';
import path from 'path';

function walk(dir) {
  let files = [];
  const list = fs.readdirSync(dir);
  list.forEach(file => {
    if (file === 'node_modules' || file === '.git' || file === 'dist') return;
    const fullPath = path.join(dir, file);
    const stat = fs.statSync(fullPath);
    if (stat && stat.isDirectory()) {
      files = files.concat(walk(fullPath));
    } else {
      files.push(fullPath);
    }
  });
  return files;
}

const allFiles = walk('.');
allFiles.forEach(file => {
  try {
    const content = fs.readFileSync(file, 'utf8');
    if (content.includes('service_role') || content.includes('SUPABASE_SERVICE_ROLE') || content.includes('postgresql://') || content.includes('postgres://')) {
      console.log(`Found secret in: ${file}`);
    }
  } catch (err) {}
});
